Update CONTRIBUTING to release workflow in CI (#1283)
informalsystems/tendermint-rs: CONTRIBUTING.md

   documentation compiles and seems up-to-date and coherent. Fix any potential
   issues here and push them to the release PR.
5. Mark the PR as **Ready for Review** and incorporate feedback on the release.

> 6. Once approved, merge the PR to `main`, tag the repo at the new version

Reviewer: ```suggestion
6. Once approved, merge the PR to `main`, create a signed tag with the new version
```
